### PR TITLE
Various improvements to the rpm package

### PR DESCRIPTION
**Pull request by haampie**
- use a multi-valued variant for crypotgraphy and berkeley-db support
- add lua, zstd and nls variants

Make gettext a build only dep when using `~nls`

Use spack's pkgconfig unconditionally instead of setting env variables
manually for it.

Add missnig iconv dep.

Doxygen is always invoked, so just make it a build dep and drop the doc
variant.

Explicitly disable the systemd inhibit plugin.

### CODE REVIEW

**Comment by haampie** [comment]
Hi @vsoch, do you want to be maintainer of this package? And would you mind reviewing?

File: var/spack/repos/builtin/packages/rpm/package.py [hunk 1]
```diff
@@ -20,67 +19,76 @@ class Rpm(AutotoolsPackage):
...
     depends_on('xz')
+    depends_on('lzma')
+    depends_on('zstd', when='+zstd')
 
     # java jar dependency analysis (already requirement for lua)
     depends_on('unzip', type='run')
 
-    # Documentation dependencies
-    depends_on('doxygen', type="build", when="+docs")
+    # Build dependencies
```
Comment (vsoch) [inline var/spack/repos/builtin/packages/rpm/package.py:86]: I'm happy to review but I don't think I am qualified to be a maintainer. For this line, why did you remove the +docs variant? Having docs build by default I don't think is what we'd want.

Comment (haampie) [reply var/spack/repos/builtin/packages/rpm/package.py:86]: I couldn't turn off the docs, it seems to build it is as part of `make all`

Comment (haampie) [reply var/spack/repos/builtin/packages/rpm/package.py:86]: Ah, it picks up `doxygen` in my PATH. And indeed they have been building docs for the past 14 years by default:

https://github.com/rpm-software-management/rpm/commit/39454472fc4ca6f776b44edc742f0a307a28ab53

"# Always build librpm docs" that line was never touched in that time frame :p

**Review by alalazo — APPROVED** [review]
